operators/pkg/controller/apmserverelasticsearchassociation/user.go

```diff
@@ -35,12 +35,18 @@ func apmUserObjectName(assocName string) string {
 // userKey is the namespaced name to identify the customer user resource created by the controller.
 func userKey(apm apmtype.ApmServer) *types.NamespacedName {
 
-	ref := apm.Spec.Output.Elasticsearch.ElasticsearchRef
-	if ref == nil {
+	ref := apm.Spec.ElasticsearchRef
+	if !ref.IsDefined() {
 		return nil
 	}
+
+	esNamespace := apm.Spec.ElasticsearchRef.Namespace
```

Comment:
Nit: we could reuse the `ref` variable: `esNamespace := ref.Namespace`. I may be a bit rigid but I would appreciate to have the same block of code with the same namings in the 2 controllers.

```go
esRef := as.Spec.Output.Elasticsearch.ElasticsearchRef
if esRef == nil {
	return nil
}

if esRef.Namespace == "" {
	// no namespace provided: default to the APM Server namespace
	esRef.Namespace = as.Namespace
}
```